File: scoring/src/main/scala/ai/h2o/sparkling/ml/params/H2OBaseMOJOParams.scala
package ai.h2o.sparkling.ml.params

import ai.h2o.sparkling.ml.models.H2OMOJOSettings
import org.apache.spark.ml.param._

trait H2OBaseMOJOParams extends Params with HasDataFrameSerializer {

  protected final val convertUnknownCategoricalLevelsToNa = new BooleanParam(
    this,
    "convertUnknownCategoricalLevelsToNa",
    "If set to 'true', the model converts unknown categorical levels to NA during making predictions.")

  protected final val convertInvalidNumbersToNa = new BooleanParam(
    this,
    "convertInvalidNumbersToNa",
    "If set to 'true', the model converts invalid numbers to NA during making predictions.")

  setDefault(
    convertUnknownCategoricalLevelsToNa -> H2OMOJOSettings.default.convertUnknownCategoricalLevelsToNa,
    convertInvalidNumbersToNa -> H2OMOJOSettings.default.convertInvalidNumbersToNa,
    dataFrameSerializer -> H2OMOJOSettings.default.dataFrameSerializer)

  def getConvertUnknownCategoricalLevelsToNa(): Boolean = $(convertUnknownCategoricalLevelsToNa)

  def getConvertInvalidNumbersToNa(): Boolean = $(convertInvalidNumbersToNa)

  def setDataFrameSerializer(fullClassName: String): this.type = set(dataFrameSerializer, fullClassName)
}

File: py-scoring/src/ai/h2o/sparkling/ml/params/H2OBaseMOJOParams.py
from ai.h2o.sparkling.ml.params.H2OTypeConverters import H2OTypeConverters
from pyspark.ml.param import *


class H2OBaseMOJOParams(Params):
    convertUnknownCategoricalLevelsToNa = Param(
        Params._dummy(),
        "convertUnknownCategoricalLevelsToNa",
        "If set to 'true', the model converts unknown categorical levels to NA during making predictions.",
        H2OTypeConverters.toBoolean())

    convertInvalidNumbersToNa = Param(
        Params._dummy(),
        "convertInvalidNumbersToNa",
        "If set to 'true', the model converts invalid numbers to NA during making predictions.",
        H2OTypeConverters.toBoolean())

    dataFrameSerializer = Param(
        Params._dummy(),
        "dataFrameSerializer",
        "A full name of a serializer used for serialization and deserialization of Spark DataFrames " +
        "to a JSON value within NullableDataFrameParam.",
        H2OTypeConverters.toString())

    ##
    # Getters
    ##
    def getConvertUnknownCategoricalLevelsToNa(self):
        return self.getOrDefault(self.convertUnknownCategoricalLevelsToNa)

    def getConvertInvalidNumbersToNa(self):
        return self.getOrDefault(self.convertInvalidNumbersToNa)

    def getDataFrameSerializer(self):
        return self.getOrDefault(self.dataFrameSerializer)

    def setDataFrameSerializer(self, value):
        return self._set(dataFrameSerializer=value)
